### PR TITLE
swift 4

### PR DESCRIPTION
**Pull request by loganwright**

### CODE REVIEW

**Comment by 0xTim** [comment]
Should `Package.resolved` files be added to the .gitignore (for libraries) given they essentially replace `Pakcage.pins`

**Comment by vzsg** [comment]
The [proposal](https://github.com/apple/swift-evolution/blob/master/proposals/0175-package-manager-revised-dependency-resolution.md) says:

> Either way, for a package which is a dependency of other packages (e.g. a library package), that package's Package.resolved file will not have any effect on its client packages.

This is the most important change from Package.pins breaking everything for clients :)

Checking in Package.resolved can now be a conscious decision about development and CI. I think it might be a good indicator if CI builds fail due to an issue in the dependencies' latest resolvable version.

---

tldr update: I mean it makes sense to ignore even Package.resolved, in a way. :)

**Comment by 0xTim** [comment]
Ah I didn't read the proposal properly! In that case, ignore me!

**Review by tanner0101 — APPROVED** [review]

**Comment by tanner0101** [comment]
looks like we'll be able to support Swift 4 without a major bump :)
